qiskit/transpiler/passes/optimization/optimize_cliffords.py

```diff
@@ -0,0 +1,85 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Combine consecutive Cliffords over the same qubits."""
+
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.quantum_info.operators import Clifford
+
+
+class OptimizeCliffords(TransformationPass):
+    """Combine consecutive Cliffords over the same qubits.
+    This serves as an example of extra capabilities enabled by storing
+    Cliffords natively on the circuit.
```

Comment:
> I think we should look to generalize CollectMultiQBlocks or similar so that we can efficiently collect and grow blocks to resynthesize, but that's a separate issue.

Agreed. This seems like some important functionality that comes up over and over.